instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/aws.go

```diff
@@ -110,6 +112,25 @@ func (m otelMiddlewares) deserializeMiddleware(stack *middleware.Stack) error {
 		middleware.Before)
 }
 
+func (m otelMiddlewares) finalizeMiddleware(stack *middleware.Stack) error {
+	return stack.Finalize.Add(middleware.FinalizeMiddlewareFunc("OTelFinalizeMiddleware", func(
+		ctx context.Context, in middleware.FinalizeInput, next middleware.FinalizeHandler) (
+		out middleware.FinalizeOutput, metadata middleware.Metadata, err error) {
+
+		// Propagate the Trace information by injecting it into the HTTP request.
+		switch req := in.Request.(type) {
+		case *smithyhttp.Request:
+			otel.GetTextMapPropagator().Inject(ctx, propagation.HeaderCarrier(req.Header))
+		default:
+			return out, metadata, fmt.Errorf("unknown transport type %T", req)
```

Comment:
```suggestion
```

It doesn't seem correct to fail all other request types here.  If this instrumentation doesn't recognize the request type it seems best to simply pass it through.

Reply:
Agreed. I have removed this.